### PR TITLE
Add constants4 directory

### PR DESCRIPTION
**Pull request by binli2337**
**Description**
Add constants4 directory for single precision floating point constants.

### CODE REVIEW

**Review by MinsukJi-NOAA — APPROVED** [review]

File: constants4/constants4.F90 [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+!!    tempc(:,:,:) = tempk(:,:,:) - TFREEZE
+!!    geopotential(:,:) = height(:,:) * grav_new
+!! @endverbatim
+
+!> @file
+!> @brief File for @ref constants_mod
+
+!> @addtogroup constants_mod
+!> @{
+module constants4_mod
```
Comment (junwang-noaa) [inline constants4/constants4.F90]: can we change the name to constantsR4_mod?

File: constants4/fmsconstants4.F90 [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+!* You should have received a copy of the GNU Lesser General Public
+!* License along with FMS.  If not, see <http://www.gnu.org/licenses/>.
+!***********************************************************************
+!> @defgroup fmsconstants FMSconstants
+!> @ingroup libfms
+!> @brief Essentially a copy of @ref constants_mod for external usage alongside
+!! @ref libfms.
+!!
+!> See @ref constants_mod for individual parameter information.
+module FMSconstants4
```
Comment (junwang-noaa) [inline constants4/fmsconstants4.F90]: The same here, may FMSconstantsR4

**Review by junwang-noaa — APPROVED** [review]

File: constants4/fmsconstants4.F90 [hunk 3]
```diff
@@ -22,11 +22,11 @@
 !! @ref libfms.
 !!
 !> See @ref constants_mod for individual parameter information.
-module FMSconstants4
+module FMSconstantsR4
 
   !> rename to not conflict with any other version vars
   use constants4_mod, version_constants => version
```
Comment (MinsukJi-NOAA) [inline constants4/fmsconstants4.F90]: Should constants4_mod be changed to constantsR4_mod here?

Comment (binli2337) [reply constants4/fmsconstants4.F90]: @MinsukJi-NOAA It is done. Thanks.

**Pull request merged**